Feat/nix

pogopaule opened the pull request:
This is my attempt to add Nix to the project.

As discussed in https://github.com/Wansmer/treesj/discussions/70, let expressions and function arguments can not be split/joined correctly. This is why I removed the half working solution for them in https://github.com/Wansmer/treesj/commit/aaeabd82fcda24d423e87c492302a55272c853f1

I am not 100% sure if you want only perfectly working splitting/joining in the plugin or not. So it is up to you if you want to revert that commit.

Wansmer commented:
Thanks for contributing!

About aaeabd8:
I fixed a problem with extra spaces (094a0df83ed6398b707f0b4460e3110f5ab76ba4) and added option for inner indent (120ec4473edb01bb1db1b73a984237bd63cc3d8e) – and node `formals` now can be added.

pogopaule commented on tests/langs/nix_spec.lua:
```diff
@@ -0,0 +1,62 @@
+    path = PATH,
+    mode = 'split',
+    lang = LANG,
+    desc = 'lang "%s", node "attrset_expression", preset default',
```
ok, done

pogopaule commented:
> Thanks for contributing!
> 
> About [aaeabd8](https://github.com/Wansmer/treesj/commit/aaeabd82fcda24d423e87c492302a55272c853f1): I fixed a problem with extra spaces ([094a0df](https://github.com/Wansmer/treesj/commit/094a0df83ed6398b707f0b4460e3110f5ab76ba4)) and added option for inner indent ([120ec44](https://github.com/Wansmer/treesj/commit/120ec4473edb01bb1db1b73a984237bd63cc3d8e)) – and node `formals` now can be added.
> 
> ```lua
>   formals = u.set_preset_for_args({
>     both = {
>       omit = { 'formal', 'ellipses' },
>     },
>     split = {
>       separator = '',
>       inner_indent = 'normal',
>     },
>     join = {
>       separator = ',',
>       space_in_brackets = true,
>     },
>   }),
> ```

Awesome! Works perfectly now.

Thanks for your support!
